clippy_utils/src/ty.rs

```diff
@@ -259,7 +260,11 @@ pub fn implements_trait_with_env_from_iter<'tcx>(
             })),
     );
 
-    debug_assert_eq!(tcx.def_kind(trait_id), DefKind::Trait);
+    debug_assert_matches!(
+        tcx.def_kind(trait_id),
+        DefKind::Trait | DefKind::TraitAlias,
+        "`DefId` must belong to a trait or trait alias"
+    );
```

Comment:
we could also just use `debug_assert!(matches!(...));` and avoid the feature, but `debug_assert_matches!` has a better ICE message (it'll print the given DefKind)

Reply:
From the looks of it, `assert_matches` is on track to getting stabilized so there isn't much of a point in doing so. I'd say keep it. The path will change though in the future, but it shouldn't cause issues (it'll be fixed within clippy in the PR that changes it)